dist/mushroom-strategy.js

```diff
@@ -156,72 +156,48 @@ function createTitleCard(title, subtitle, offService, onService, iconOff, iconOn
  *
  * Double-tapping opens the more-info popup of home assistant, unless given a custom double-tap configuration.
  *
- * @param {Set<hassEntity>} entities Registered Hass entities.
+ * @param {hassEntity[]} entities Registered Hass entities.
  * @param {entityConfig[]} entity_config Custom card-configurations for an entity on a view.
  * @param {Object} defaultCard Default card-configuration for the entities on a view.
  * @param {Object} titleCard Optional title card.
  * @param {Object} doubleTapActionConfig Custom configuration for the card's double-tap action.
  *
  * @return {Object[]} Array of view cards.
+ * @todo: Apply a filter to entities instead of iterating it manually.
  */
-function createViewCards(entities, entity_config, defaultCard, titleCard, doubleTapActionConfig) {
-  const platformCards = [];
+function createViewCards(entities, entity_config, defaultCard, titleCard, doubleTapActionConfig = null) {
+  const viewCards = [];
 
-  if (titleCard != null) {
-    platformCards.push(titleCard);
+  if (titleCard) {
+    viewCards.push(titleCard);
   }
 
-  let doubleTapAction;
-
-  entitiesLoop:
-      for (const entity of entities) {
-        // Entity config does not exist then push default card, otherwise loop to find matching entity
-        if (entity_config == null) {
-          if (doubleTapActionConfig != null) {
-            doubleTapAction = {
-              double_tap_action: {
-                target: {
-                  entity_id: entity.entity_id,
-                },
-                ...doubleTapActionConfig,
-              },
-            };
-          }
-
-          platformCards.push({
-            entity: entity.entity_id,
-            ...defaultCard,
-            ...doubleTapAction,
-          });
-        } else {
-          for (const config of entity_config) {
-            if (entity.entity_id == config.entity) {
-              platformCards.push({...config});
+  for (const entity of entities) {
+    if ((entity.entity_id in (entity_config ?? {}))) {
```

Comment:
Not sure why but `entity_config` entries are not being used. I belive the condition is never met.

Reply:
You mean you pass a value to parameter `entity_config`, but line 178 isn't executed?

I think I've mistaken `entity_config` to be an object of objects, while it is an array of objects.
At the moment I don't have access to my Home Assistant instance. Could you try the following in the meanwhile?

```suggestion
    if ((entity_config ?? []).find(config => config.entity === entity.entity_id)) {
```

I will apply a fix, this evening ;).

Reply:
Yes, that is what I mean.
The fix works but it pushes the entity_id only and not the entire card. I get an error.

![image](https://user-images.githubusercontent.com/39784853/231499961-c738467b-f23a-4f21-8713-693659ca9136.png)

Are there some docs where I can read some documentation for array and set methods such as `.filter` and `.find`?

Reply:
> Are there some docs where I can read some documentation for array and set methods such as `.filter` and `.find`?

Of course, there are many resources available online.
I like to consult the [MDN Web Docs](https://developer.mozilla.org/en-US/) of Mozilla.